File: 0007. Reverse Integer/Solution.py
# Author        : tarlansoltanov
# Title         : Reverse Integer
# ID            : 0007
# URL           : https://leetcode.com/problems/reverse-integer/
# Difficulty    : Medium
# Language      : Python


class Solution:
    def reverse(self, x: int) -> int:
        if abs(x) <= 9:
            return x
        flag = x // abs(x)
        x, ans  = abs(x), 0
        while x > 0:
            ans *= 10
            ans += x % 10
            x //= 10
        if ans > 2**31 - 1:
            return 0
        return flag * ans

File: 0007. Reverse Integer/Solution.cpp
// Author        : tarlansoltanov
// Title         : Reverse Integer
// ID            : 0007
// URL           : https://leetcode.com/problems/reverse-integer/
// Difficulty    : Medium
// Language      : C++


class Solution {
public:
    long long reverse(long long x) {
        if(fabs(x) <= 9){
            return x;
        }

        long long flag = x / fabs(x), num = fabs(x), ans = 0;

        while(num > 0){
            if(ans > (pow(2, 31) - 1) / 10){
                return 0;
            }
            ans *= 10;
            ans += num % 10;
            num /= 10;
        }

        return flag * ans;
    }
};